core/src/test/java/org/apache/iceberg/rest/TestHTTPClient.java

```diff
@@ -133,6 +134,21 @@ public void testDynamicHttpRequestInterceptorLoading() {
     assertThat(((TestHttpRequestInterceptor) interceptor).properties).isEqualTo(properties);
   }
 
+  @Test
+  public void testHttpClientGetConnectionConfig() {
```

Comment:
A better way to test is using mockServer's delay response to test timeout. https://stackoverflow.com/questions/39638974/mockwebserver-response-delay-testing-timeouts

Reply:
+1 to having a test case that functionally exercises the timeout behavior if possible.

If we do that though we should make sure we set the intended timeout low enough that we're not noticeably slowing down the unittest (we'd expect the configured timeout to be much shorter than the "delay" we make the mockResponse use).